### DISCUSSION TITLE
Extending the rules engine with custom plug-ins

### DISCUSSION THREAD

**Discussion by rfennell** [tfsaggregator/aggregator-cli#247] [Q&A]
I have a potential project where there is a need to perform two-way synchronisation of Azure DevOps WI with ServiceNow tickets. Has extending TFSAggregator been considered an option for this type of workload, i.e., adding a ServiceNow API plug-in to the rules engine?

As far as I can see it my other options are:

A completely bespoke solution, that I think would end up looking a lot like this tool, but with a hard coded rules engine
Using Zapier or similar.

**Comment by giuliov** [comment]
Interesting idea @rfennell . I am catching up with backlog, so I hope I can get back to you with some ideas.
